sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -129,8 +129,17 @@ trait SchemaRelationProvider {
  * Implemented by objects that can produce a streaming [[Source]] for a specific format or system.
  */
 trait StreamSourceProvider {
+
+  /** Returns the name and schema of the source that can be used to continually read data. */
+  def sourceSchema(
+      sqlContext: SQLContext,
+      schema: Option[StructType],
+      providerName: String,
+      parameters: Map[String, String]): (String, StructType)
+
   def createSource(
       sqlContext: SQLContext,
+      sourceId: Long,
```

Comment:
I thought the goal was to have all the data in the same location.  With this API everyone needs to duplicate the checkpoint location resolution logic.

Note that if you want a unique identifier the path also qualifies.